tests/python_package_test/test_engine.py

```diff
@@ -1068,6 +1068,39 @@ def test_cvbooster():
     assert ret < 0.15
 
 
+def test_cvbooster_save_load():
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'verbose': -1,
+    }
+    nfold = 3
+    lgb_train = lgb.Dataset(X_train, y_train)
+
+    def predict(cv_booster):
+        return np.array(cv_booster.predict(X_test))
+
+    cv_res = lgb.cv(params, lgb_train,
+                    num_boost_round=25,
+                    nfold=nfold,
+                    callbacks=[lgb.early_stopping(stopping_rounds=5)],
+                    return_cvbooster=True)
+    cvbooster = cv_res['cvbooster']
+
+    ret_origin = predict(cvbooster)
+    other_ret = []
+
+    cvbooster.save_model('lgb.model')
+
+    other_ret.append(predict(lgb.CVBooster(model_file='lgb.model')))
+    other_ret.append(predict(lgb.CVBooster().model_from_string(cvbooster.model_to_string())))
+
+    for ret in other_ret:
+        np.testing.assert_array_equal(ret_origin, ret)
```

Comment:
```suggestion
    preds = predict(cvbooster)

    cvbooster.save_model('lgb.model')
    preds_from_file = predict(lgb.CVBooster(model_file='lgb.model')))
    preds_from_string =  predict(lgb.CVBooster().model_from_string(cvbooster.model_to_string())))

    np.testing.assert_array_equal(preds, preds_from_file)
    np.testing.assert_array_equal(preds, preds_from_string)
```

I think the indirection added by using this empty list and then a `for` loop over it makes it a bit difficult to understand what is being tested here. Will you please consider a rewrite like this?

Reply:
Also, to ensure that files written by these tests are automatically cleaned up, please use the `pytest` `tmp_path` fixture.

For example:

https://github.com/microsoft/LightGBM/blob/6b56a90cd1324a6dbac2afa0a352c9355b0dc3cf/tests/python_package_test/test_engine.py#L2922

https://github.com/microsoft/LightGBM/blob/6b56a90cd1324a6dbac2afa0a352c9355b0dc3cf/tests/python_package_test/test_engine.py#L3023-L3024

Reply:
@jameslamb 
I wrote it with reference to the other test case (`test_save_load_copy_pickle`), but it seems that your suggestion is more concise and readable. Thanks ;)